crashes on compilable code: NoDenotation.owner

scalac can compile the following code; but dotc crashes.

$ cat abc.scala

```
package util.util
trait Bootable {
}
```

$ scalac abc.scala

$ dotc abc.scala

```
-- [E008] Member Not Found Error: abc.scala ----------------------------------------------------------------------------
1 |package util.util
  |        ^^^^^^^^^
  |        value `util` is not a member of util.type - did you mean `util.Try`?

exception occurred while typechecking abc.scala

exception occurred while compiling abc.scala
Exception in thread "main" java.lang.AssertionError: NoDenotation.owner
    at dotty.tools.dotc.core.SymDenotations$NoDenotation.owner(SymDenotations.scala:1868)
    at dotty.tools.dotc.core.SymDenotations$SymDenotation.valRef(SymDenotations.scala:1112)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedPackageDef$1.apply(Typer.scala:1347)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedPackageDef$1.apply(Typer.scala:1337)
    at dotty.tools.dotc.util.Stats$.track(Stats.scala:35)
    at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:1337)
    at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:1468)
    at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:1480)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1490)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1488)
    at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:132)
    at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:57)
    at dotty.tools.dotc.typer.Typer.typed(Typer.scala:1488)
    at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:1540)
    at dotty.tools.dotc.typer.FrontEnd$$anonfun$typeCheck$1.apply$mcV$sp(FrontEnd.scala:50)
    at dotty.tools.dotc.typer.FrontEnd.monitor(FrontEnd.scala:24)
    at dotty.tools.dotc.typer.FrontEnd.typeCheck(FrontEnd.scala:48)
    at dotty.tools.dotc.typer.FrontEnd$$anonfun$runOn$3.apply(FrontEnd.scala:72)
    at dotty.tools.dotc.typer.FrontEnd$$anonfun$runOn$3.apply(FrontEnd.scala:72)
    at scala.collection.immutable.List.foreach(List.scala:381)
    at dotty.tools.dotc.typer.FrontEnd.runOn(FrontEnd.scala:72)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1$$anonfun$apply$mcV$sp$1.apply(Run.scala:75)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1$$anonfun$apply$mcV$sp$1.apply(Run.scala:72)
    at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
    at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply$mcV$sp(Run.scala:72)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply(Run.scala:66)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply(Run.scala:66)
    at dotty.tools.dotc.util.Stats$.monitorHeartBeat(Stats.scala:76)
    at dotty.tools.dotc.Run.compileUnits(Run.scala:66)
    at dotty.tools.dotc.Run.compileSources(Run.scala:63)
    at dotty.tools.dotc.Run.compile(Run.scala:47)
    at dotty.tools.dotc.Driver.doCompile(Driver.scala:26)
    at dotty.tools.dotc.Driver.process(Driver.scala:124)
    at dotty.tools.dotc.Driver.process(Driver.scala:93)
    at dotty.tools.dotc.Driver.process(Driver.scala:105)
    at dotty.tools.dotc.Driver.main(Driver.scala:132)
    at dotty.tools.dotc.Main.main(Main.scala)
```

--- a/src/dotty/tools/dotc/core/Mode.scala
+++ b/src/dotty/tools/dotc/core/Mode.scala
@@ -93,4 +93,9 @@ object Mode {
   val ReadPositions = newMode(16, "ReadPositions")
 
   val PatternOrType = Pattern | Type
+  
+  /** We are elaborating the fully qualified name of a package clause.
+   *  In this case, identifiers should never be imported.
+   */
+  val InPackageClauseName = newMode(17, "InPackageClauseName")
 }

--- a/src/dotty/tools/dotc/typer/Typer.scala
+++ b/src/dotty/tools/dotc/typer/Typer.scala
@@ -95,6 +95,7 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
   def typedIdent(tree: untpd.Ident, pt: Type)(implicit ctx: Context): Tree = track("typedIdent") {
     val refctx = ctx
     val name = tree.name
+    val noImports = ctx.mode.is(Mode.InPackageClauseName)
 
     /** Method is necessary because error messages need to bind to
      *  to typedIdent's context which is lost in nested calls to findRef
@@ -240,7 +241,8 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
 
       /** Would import of kind `prec` be not shadowed by a nested higher-precedence definition? */
       def isPossibleImport(prec: Int)(implicit ctx: Context) =
-        prevPrec < prec || prevPrec == prec && (prevCtx.scope eq ctx.scope)
+        !noImports && 
+        (prevPrec < prec || prevPrec == prec && (prevCtx.scope eq ctx.scope))
 
       @tailrec def loop(implicit ctx: Context): Type = {
         if (ctx.scope == null) previous
@@ -1335,7 +1337,7 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
   }
 
   def typedPackageDef(tree: untpd.PackageDef)(implicit ctx: Context): Tree = track("typedPackageDef") {
-    val pid1 = typedExpr(tree.pid, AnySelectionProto)
+    val pid1 = typedExpr(tree.pid, AnySelectionProto)(ctx.addMode(Mode.InPackageClauseName))
     val pkg = pid1.symbol
     val packageContext =
       if (pkg is Package) ctx.fresh.setOwner(pkg.moduleClass).setTree(tree)

--- a/tests/pos/i1638.scala
+++ b/tests/pos/i1638.scala
@@ -1,0 +1,3 @@
+package util.util
+
+class C

